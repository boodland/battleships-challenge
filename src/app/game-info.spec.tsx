import { render } from '@testing-library/react';

import GameInfo from './game-info';

const mocks = vi.hoisted(() => {
  return {
    hasGameFinished: vi.fn().mockReturnValue(false),
    getShipsInfo: vi.fn().mockReturnValue([
      {type: 'Type 1', isShunk: false},
      {type: 'Type 2', isShunk: false},
    ]),
  }
})

vi.mock('./utils', () => {
  return { hasGameFinished: mocks.hasGameFinished, getShipsInfo: mocks.getShipsInfo};
});

describe('GameInfo', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<GameInfo ships={[]} shots={[]} />);
    expect(baseElement).toBeTruthy();
  });

  it('should display ships type', () => {
    const { container } = render(<GameInfo ships={[]} shots={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('should display ships type crossed if is shunk', () => {
    const shipsInfoShunk = [
      {type: 'Type 1', isShunk: true},
      {type: 'Type 2', isShunk: true},
    ]
    mocks.getShipsInfo.mockReturnValue(shipsInfoShunk)
    const { container } = render(<GameInfo ships={[]} shots={[]} />);
    expect(container).toMatchSnapshot();
  });

  it('should display Game Finished! if game has finished', () => {
    mocks.hasGameFinished.mockReturnValue(true);
    const { container } = render(<GameInfo ships={[]} shots={[]} />);
    expect(container).toMatchSnapshot();
  });
});
