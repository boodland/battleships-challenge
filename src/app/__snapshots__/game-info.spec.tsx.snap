// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameInfo > should display Game Finished! if game has finished 1`] = `
<div>
  <div
    class="_text_3ba226"
  >
    <div
      class="_shunk_3ba226"
    >
      Type 1
    </div>
    <div
      class="_shunk_3ba226"
    >
      Type 2
    </div>
    Game Finished!
  </div>
</div>
`;

exports[`GameInfo > should display ships type 1`] = `
<div>
  <div
    class="_text_3ba226"
  >
    <div
      class=""
    >
      Type 1
    </div>
    <div
      class=""
    >
      Type 2
    </div>
  </div>
</div>
`;

exports[`GameInfo > should display ships type crossed if is shunk 1`] = `
<div>
  <div
    class="_text_3ba226"
  >
    <div
      class="_shunk_3ba226"
    >
      Type 1
    </div>
    <div
      class="_shunk_3ba226"
    >
      Type 2
    </div>
  </div>
</div>
`;
